File: DSA-for-Beginners/03_Recursion/01_Factorial.cpp
#include <iostream>

int factorial(int n) {
    // Base case
    if (n <= 1) {
        return 1;
    }

    // Recursive step
    return n * factorial(n-1);
}

int main() {
    int n = 5;
    int result = factorial(n);
    printf("%d! = %d\n", n, result);

    return 0;
}

File: DSA-for-Beginners/03_Recursion/01_Factorial.py
def factorial(n):
    # Base case
    if n <= 1:
        return 1
    
    # Recursive step
    return n * factorial(n - 1)

def main():
    n = 5
    result = factorial(n)
    print(f"{n}! = {result}")

if __name__ == "__main__":
    main()